main.go

```diff
@@ -201,6 +202,15 @@ func main() {
 			return
 		}
 
+		if cfg.auth.Authentication.Header.Enabled {
+			forwardedFor, err := getOutboundIP(upstreamURL.Host)
+			if err != nil {
+				glog.Errorf("Failed to find outbound IP for upstream %s: %v", upstreamURL.Host, err)
+				return
```

Comment:
if no outbound IP could be found, we are returning from the http handler function immediately. We should at least probably return a 503 here?

Reply:
Or... Don't return. The proxy call should also fail.